clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -553,3 +748,20 @@ void HLSLExternalSemaSource::CompleteType(TagDecl *Tag) {
     return;
   It->second(Record);
 }
+
+static FunctionDecl *lookupBuiltinFunction(Sema &S, StringRef Name) {
+  IdentifierInfo &II =
+      S.getASTContext().Idents.get(Name, tok::TokenKind::identifier);
+  DeclarationNameInfo NameInfo =
+      DeclarationNameInfo(DeclarationName(&II), SourceLocation());
+  LookupResult R(S, NameInfo, Sema::LookupOrdinaryName);
+  // AllowBuiltinCreation is false but LookupDirect will create
+  // the builtin when searching the global scope anyways...
+  S.LookupName(R, S.getCurScope());
+  // FIXME: If the builtin function was user-declared in global scope,
```

Comment:
Wouldn't that require doing so here inside HLSLExternalSemaSource?  How else would this find something before we even start parsing the built-in HLSL header, let alone the source file?

Reply:
Since this is just moving a pre-existing function, feel free to ignore this comment for the purpose of completing the PR.  I'm just wondering out loud.